pkg/proto/protobuf_converters.go

```diff
@@ -60,6 +60,217 @@ func SignedTxFromProtobuf(data []byte) (Transaction, error) {
 	return res, nil
 }
 
+func balancesFromProto(scheme Scheme, txSnapshotProto *g.TransactionStateSnapshot, res *[]AtomicSnapshot) error {
+	for _, balance := range txSnapshotProto.Balances {
+		if balance.Amount.AssetId == nil {
+			var sn WavesBalanceSnapshot
+			err := sn.FromProtobuf(scheme, balance)
+			if err != nil {
+				return err
+			}
+			*res = append(*res, sn)
+		} else {
+			var sn AssetBalanceSnapshot
+			err := sn.FromProtobuf(scheme, balance)
+			if err != nil {
+				return err
+			}
+			*res = append(*res, sn)
+		}
+	}
+	return nil
+}
+func leaseBalancesFromProto(scheme Scheme, txSnapshotProto *g.TransactionStateSnapshot, res *[]AtomicSnapshot) error {
```

Comment:
I see that this function appends new snapshot to the `res`.
On my mind the more convenient function signature is:
```go
func appendBalancesFromProto(
	res []AtomicSnapshot,
	scheme Scheme,
	balances []*g.TransactionStateSnapshot_Balance,
) ([]AtomicSnapshot, error) {
	// ...
	return res, nil
}
```
Issue can be applied for all similar functions.

Reply:
it is need for linter issues (funclen)

Reply:
Maybe better introduce an `AtomicSnapshots` type and make this function and other functions like this the methods of this type?